pkg/apis/hco/v1alpha1/hyperconverged_webhook.go

```diff
@@ -0,0 +1,106 @@
+package v1alpha1
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"path/filepath"
+
+	hcoutil "github.com/kubevirt/hyperconverged-cluster-operator/pkg/util"
+	admissionregistrationv1 "k8s.io/api/admissionregistration/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/webhook"
+)
+
+const (
+	WebhookPort     = 4343
+	WebhookCertDir  = "/apiserver.local.config/certificates"
+	WebhookCertName = "apiserver.crt"
+	WebhookKeyName  = "apiserver.key"
+)
+
+var hcolog = logf.Log.WithName("hyperconverged-resource")
+
+func (r *HyperConverged) SetupWebhookWithManager(mgr ctrl.Manager) error {
+	// Make sure the certificates are mounted, this should be handled by the OLM
+	certs := []string{filepath.Join(WebhookCertDir, WebhookCertName), filepath.Join(WebhookCertDir, WebhookKeyName)}
+	for _, fname := range certs {
+		if _, err := os.Stat(fname); err != nil {
+			hcolog.Info("CSV certificates were not found, skipping webhook initialization")
+			return nil
+		}
+	}
+
+	// The OLM limits the webhook scope to the namespaces that are defined in the OperatorGroup,
+	// by setting namespaceSelector in the ValidatingWebhookConfiguration.  We would like our webhook to intercept
+	// CREATE from all namespaces, and fail them if they're not in the correct namespace for HCO.
+	// To do that, we need to find our ValidatingWebhookConfiguration and reset the namespaceSelector that was
+	// defined by OLM.  Luckily, the OLM does not reconcile the ValidatingWebhookConfiguration
+
+	vwcList := &admissionregistrationv1.ValidatingWebhookConfigurationList{}
+	err := mgr.GetAPIReader().List(context.TODO(), vwcList, client.MatchingLabels{"olm.webhook-description-generate-name": hcoutil.HcoValidatingWebhook})
```

Comment:
context should represent a single flow; e.g. a specific HTTP call. Please consider to use the same context in all the calls - create a context at the beginning of the function and use it in all the calls to K8s.

Reply:
Done